state/redis/redis.go

```diff
@@ -68,28 +69,28 @@ func (r *StateStore) Init(metadata state.Metadata) error {
 		return err
 	}
 
-	ctx := context.Background()
-	opts := redisconn.Opts{
-		DB:       0,
-		Password: redisCreds.Password,
-	}
-	conn, err := redisconn.Connect(ctx, redisCreds.Host, opts)
+	var tlsConfig *tls.Config
+	err = json.Unmarshal(b, tlsConfig)
 	if err != nil {
 		return err
 	}
 
-	r.client = &redis.SyncCtx{
-		S: conn,
+	opts := redis.Options{
+		DB:        0,
+		Password:  redisCreds.Password,
+		TLSConfig: tlsConfig,
 	}
 
+	r.client = redis.NewClient(&opts)
+
```

Comment:
You might need to check for connection in init() through client.Ping().Result().

Reply:
Added the check now.